segmentstore/server/src/test/java/io/pravega/segmentstore/server/logs/ContainerMetadataUpdateTransactionTests.java

```diff
@@ -347,6 +352,102 @@ private void testWithAttributes(Function<Collection<AttributeUpdate>, Operation>
                 expectedValues, metadata.getStreamSegmentMetadata(SEGMENT_ID));
     }
 
+    private void testWithAttributesByReference(Function<Collection<AttributeUpdate>, Operation> createOperation) throws Exception {
+        final UUID referenceAttributeId = UUID.randomUUID();
+        final UUID attributeNoUpdate = UUID.randomUUID();
+        final UUID attributeAccumulate = UUID.randomUUID();
+        final UUID attributeReplaceIfEquals = UUID.randomUUID();
+        final long initialAttributeValue = 1234567;
+
+        UpdateableContainerMetadata metadata = createMetadata();
+        metadata.getStreamSegmentMetadata(SEGMENT_ID)
+                .updateAttributes(ImmutableMap.of(
+                        referenceAttributeId, initialAttributeValue,
+                        attributeAccumulate, initialAttributeValue,
+                        attributeNoUpdate, 123L,
+                        attributeReplaceIfEquals, initialAttributeValue));
+
+        val txn = createUpdateTransaction(metadata);
+
+        // Update #1 (Value-by-reference).
+        Collection<AttributeUpdate> attributeUpdates = Arrays.asList(
+                new AttributeUpdateByReference(
+                        attributeAccumulate,
+                        AttributeUpdateType.Accumulate,
+                        new AttributeReference<>(referenceAttributeId, v -> v + 2)), // PrevValue + AttributeReference + 2
+                new AttributeUpdateByReference(
+                        attributeReplaceIfEquals,
+                        AttributeUpdateType.ReplaceIfEquals,
+                        new AttributeReference<>(referenceAttributeId, v -> v + 3), initialAttributeValue)); // AttributeReference + 3
+
+        Map<UUID, Long> expectedValues = ImmutableMap.of(
+                referenceAttributeId, initialAttributeValue,
+                attributeAccumulate, initialAttributeValue + initialAttributeValue + 2,
+                attributeNoUpdate, 123L,
+                attributeReplaceIfEquals, initialAttributeValue + 3);
+
+        Operation op = createOperation.apply(attributeUpdates);
+        txn.preProcessOperation(op);
+        txn.acceptOperation(op);
+
+        // Verify result.
+        verifyAttributeUpdates("after acceptOperation (1)", txn, attributeUpdates, expectedValues);
+
+        // Update #2 (Value-by-reference).
```

Comment:
**Minor**: I think `value-by-reference` and `id-by-reference` are making references to `pass-by-value` and `pass-by-reference` when referring to parameter passing. Do you think we can reuse that terminology, which is more familiar, or is it going to be confusing?